observer/include/observer/filters/geometry/GeometryFilter.h

```diff
@@ -0,0 +1,23 @@
+//
+// Created by rolf on 26-10-20.
+//
+
+#ifndef RTT_GEOMETRYFILTER_H
+#define RTT_GEOMETRYFILTER_H
+
+
+#include <roboteam_proto/messages_robocup_ssl_geometry.pb.h>
+class GeometryFilter {
+public:
+    bool process(const proto::SSL_GeometryData& geometryData);
+    bool receivedFirstGeometry() const;
+    proto::SSL_GeometryData getGeometry() const;
+private:
+    std::string lastGeometryString;
+    proto::SSL_GeometryData combinedGeometry;
+    std::map<unsigned int,proto::SSL_GeometryCameraCalibration> cameras;
```

Comment:
Why don't we use std::vector here for O(1) access?

Reply:
I mean, we can, but I thought the map was clearer / more logical  and takes up less memory in general. We only need to access the camera's by ID anyways and this map has very little elements. Otherwise, we need to define an upper limit on the amount of camera's on the field. 

Additionally, we don't always know what ID SSL-vision will give the camera's, it might give very high ID's for whatever reason in the future. I think this is more stable and reliable (at the cost of some extremely minor performance).

Reply:
Unless, you want to search over the vector, which is O(n).

Reply:
Oh okay I just thought it would be ID's starting at 0, incrementing, nevermind then.